dsp/window/window_test.go

```diff
@@ -8,170 +8,317 @@ import (
 	"fmt"
 	"testing"
 
+	"gonum.org/v1/gonum/cmplxs"
 	"gonum.org/v1/gonum/floats"
-	"gonum.org/v1/gonum/floats/scalar"
 )
 
+const tol = 1e-6
 const (
 	gaussWin = iota
 	tukeyWin
 )
 
-var windowTests = []struct {
-	name    string
-	fn      func([]float64) []float64
-	fnCmplx func([]complex128) []complex128
-	want    []float64
+// these types describe zero, mono, and multi parameter windowing
+// functions for both real and complex data
+//
+// when looking at the truth tables, it is helpful to know that
+// Go treats []complex128{1} as [1+0i].  More specifically, a variable which
+// is of type complex128 to which you assign an untyped constant has a zero
+// imaginary part.
+
+type transformFunc func([]float64) []float64
+type transformFuncComplex func([]complex128) []complex128
```

Comment:
Please delete these and use the untyped function declarations in the structs.

Reply:
Why?

Reply:
Locality. Name indirection doesn't help here, it just adds words.